ImmuDB4Net/ConnectionPool.cs

```diff
@@ -21,82 +21,227 @@ namespace ImmuDB;
 public interface IConnectionPool
 {
     int MaxConnectionsPerServer { get; }
-    IConnection Acquire(ImmuClient client);
-    void Release(ImmuClient client);
+    IConnection Acquire(ConnectionParameters cp);
+    void Release(IConnection con);
     Task Shutdown();
 }
 
+
+
 public class RandomAssignConnectionPool : IConnectionPool
 {
     public int MaxConnectionsPerServer { get; private set; }
+    public TimeSpan TerminateIdleConnectionTimeout { get; private set; }
+    public TimeSpan IdleConnectionCheckInterval { get; private set; }
     private Random random = new Random(Environment.TickCount);
     private ReleasedConnection releasedConnection;
```

Comment:
We don't need releasedConnection here anymore, it's handled at the ImmuClient level now.

Reply:
fixed